core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2379,8 +2382,13 @@ class SparkContext(config: SparkConf) extends Logging {
                 " parameter from breaking Spark's ability to find a valid constructor.")
           }
         }
-        listenerBus.addListener(listener)
-        logInfo(s"Registered listener $className")
+        logInfo(s"listener $className created")
+        listener
+      }
+      if (extraListeners.nonEmpty) {
+        val group = new FixGroupOfListener(extraListeners, "extraListeners")
```

Comment:
> First, I think that modifying the existing addListener method is a bad idea. It will impact a lot the code.

That's why overloaded methods exist.

> But I have 2 kind of group of listeners

I don't think there's really a distinction between the two types of groups you mention. The "UI group" is just a modifiable group that you don't modify after it's been created.